src/Classes/ServiceAPI/MyRadio_Show.php

```diff
@@ -1028,18 +1028,31 @@ public static function getMostListened($date = 0)
 
     /**
      * Searches searchable *text* metadata for the specified value. Does not work for image metadata.
+     * if $q is set, then $path_query must be set to "search"
      *
      * @todo effective_from/to not yet implemented
      *
-     * @param string $query          The query value.
+     * @param string $path_query     The query value encoded in the path (DEPRECATED).
+     * @param string $q              The query value as a query string. to use this, $path_query must be set to "search"
      * @param array  $string_keys    The metadata keys to search
      * @param int    $effective_from UTC Time to search from.
      * @param int    $effective_to   UTC Time to search to.
      *
      * @return array The shows that match the search terms
      */
-    public static function searchMeta($query, $string_keys = null, $effective_from = null, $effective_to = null)
+    public static function searchMeta($path_query, $q="", $string_keys = null, $effective_from = null, $effective_to = null)
     {
+        if($path_query != "search" && $q != "") {
+            throw new MyRadioException(
+                "the path_query must be set to 'search' if q is set in the query string",
+                400
+            );
+        }
+        if ($path_query == "search" && $q != ""){
+            $query = $q;
+        } else {
+            $query = $path_query;
+        }
```

Comment:
what is the point of adding a parameter if in both cases $query gets set to one or the other?

Reply:
this is because of myradio weirdness. in the service api when called the first arg is gotten from the path section of the url. subsequent args are gotten from query parameters. to get a query parameter you need a second arg. we set query based on whether they are using query parameters or path to maintain backwards compatibility